File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] numbers) {
        int[] answer = new int[numbers.Length];
        
        Stack<int> stack = new Stack<int>();
        stack.Push(0);
        
        for(int i = 1; i < numbers.Length; i++)
        {
            while(stack.Count > 0 && numbers[stack.Peek()] < numbers[i])
            {
                answer[stack.Pop()] = numbers[i];                
            }
            stack.Push(i);
        }
        
        while(stack.Count > 0)
        {
            answer[stack.Pop()] = -1;
        }
        return answer;
    }
}

File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer(numbers.size(), -1);
    stack<int> stk;
    
    for(int i = 0; i < numbers.size(); i++)
    {
        while(stk.size() && numbers[stk.top()] < numbers[i])
        {
            answer[stk.top()] = numbers[i];
            stk.pop();
        }            
        stk.push(i);
    }
    return answer;
}